doc/crypto/api/keys/management.rst

```diff
@@ -497,30 +496,38 @@ When creating a key, the attributes for the new key are specified in a `psa_key_
+    *   An implementation-provided key can be fully defined by the implementation, with a fixed type, size, and policy.
+        The call to `psa_attach_key()` needs to specify the location and label of the key, and a macthing policy, in order to obtain a key id.
 
-    Calling `psa_destroy_key()` with a key identifier returned by `psa_attach_key()` will remove the key identifer and policy from the key store, but the original key material remains within the implementation.
+    Calling `psa_destroy_key()` with a key identifier returned by `psa_attach_key()` will remove the key identifer and policy from the key store, but any implementation-provided key material remains within the implementation.
+    A subsequent call to `psa_attach_key()` with the same parameters will return a new key identifier for the same key.
 
-    It is :scterm:`implementation defined` whether the same key material can be attached to multiple key identifiers concurrently.
+    It is :scterm:`implementation defined` whether the same implementation-provided key can be attached to multiple key identifiers concurrently.
 
     .. note::
 
-        This function is intended for scenarios where key material is provisioned outside the |API|, and the application needs to use such keys within the API framework.
+        This function is intended for scenarios where key material is provided outside the |API|, and the application needs to use such keys within the |API| framework.
 
-        Implementations may impose restrictions on which keys can be registered, depending on their storage architecture and security policies.
-
-        The function does not allow registering keys that are not already present in the implementation's storage.
+        The function does not allow registering keys that are not already provided by the implementation.
         To import new key material, use `psa_import_key()`.
 
-        The function does not modify the key material; it only registers it under the specified identifier and attributes.
+        Although the implementation verifies that the application-supplied attributes are compatible with the implementation-provided key; it is the application's responsibility to ensure correctness for attributes that are provided by the implementation.
 
         To create a persistent key from pre-existing key material, the implementation might permit a key returned by `psa_attach_key()` to be copied to a persistent key using `psa_copy_key()`.
 
+    .. admonition:: Implementation note
+
+        Implementations may impose restrictions on which keys can be registered, depending on their storage architecture and security policies.
+
+        The behavior of a call `psa_attach_key()` with a persistent key-lifetime might be specified in a future version of the |API|.
+        Such a call must result in a :code:`PSA_ERROR_INVALID_ARGUMENT` error in an implementation of version |docversion| of the |API|.
```

Comment:
Stating that the call *must* return this error means that if we define the behavior in 1.5, it'll be impossible for an implementation to claim both 1.5 and 1.4 compatibility. I don't think that's desirable. Also, an implementation might want to return a different error code, for example an implementation without persistent storage might return `PSA_ERROR_NOT_SUPPORTED` for any access to persistent storage. So I think this should be a SHOULD, or reworded to say that implementations should return an error rather than defining implementation-specific behavior.

Reply:
I think that was your original wording in a comment - but we should relax this.

Reply:
Here's my proposed alternative wording:

> The behavior of a call `psa_attach_key()` with a persistent key-lifetime might be specified in a future version of the Crypto API. At present, it is recommended that such a call returns an error status, and does not provide implementation-specific behavior.